code/modules/research/destructive_analyzer.dm

```diff
@@ -81,6 +89,11 @@ Note: Must be placed within 3 tiles of the R&D Console
 		var/list/food = thing.GetDeconstructableContents()
 		for(var/obj/item/innerthing in food)
 			destroy_item(innerthing, TRUE)
+	if(can_research_items && !is_type_in_typecache(thing.type, researched_items))
```

Comment:
why don't you just store the type of the item and check istype

/obj/item/subtype will return TRUE on istype(item, /obj/item)

also this kinda sucks since you can just deconstruct the furthest down item in the hierarchy and then work your way up.

Also you can block off lots of items by researching something high up